front/src/modules/simulationResult/components/ChartHelpers/drawElectricalProfile.ts

```diff
@@ -0,0 +1,278 @@
+import { pointer } from 'd3-selection';
+import i18n from 'i18n';
+import { Chart } from 'reducers/osrdsimulation/types';
+import { ElectricalConditionSegment, DrawingKeys } from 'applications/operationalStudies/consts';
+import { getElementWidth } from './ChartHelpers';
+
+const drawElectricalProfile = (
+  chart: Chart,
+  classes: string,
+  electricalConditionSegment: ElectricalConditionSegment,
+  groupID: string,
+  keyValues: DrawingKeys,
+  rotate: boolean,
+  isStripe: boolean,
+  isIncompatible: boolean,
+  id: string
+) => {
+  const ELECTRIFIED = 'Electrified';
+  const NEUTRAL = 'Neutral';
+  const electrificationType = electricalConditionSegment.electrification.object_type;
+  const electrificationProfile =
+    electrificationType === ELECTRIFIED
+      ? electricalConditionSegment.electrification.profile
+      : undefined;
+  const electrificationMode =
+    electrificationType === ELECTRIFIED
+      ? electricalConditionSegment.electrification.mode
+      : undefined;
+
+  // Get the different axis values based on rotation
+  // x
+  const xValues = {
+    start: electricalConditionSegment[`${keyValues[rotate ? 1 : 0]}_start`],
+    middle: electricalConditionSegment[`${keyValues[rotate ? 1 : 0]}_middle`],
+    end: electricalConditionSegment[`${keyValues[rotate ? 1 : 0]}_end`],
+  };
+  // y
+  const yValues = {
+    start: electricalConditionSegment[`${keyValues[rotate ? 0 : 1]}_start`],
+    middle: electricalConditionSegment[`${keyValues[rotate ? 0 : 1]}_middle`],
+    end: electricalConditionSegment[`${keyValues[rotate ? 0 : 1]}_end`],
+  };
```

Comment:
Nice idea ! Just a suggestion but as it's also used in drawPowerRestriction, maybe you could make a function to get these values and put it in chartHelpers.ts for example. The suggestion below can probably be improved.
```suggestion
 export const getValues = (
  segment: ElectricalConditionSegment | PowerRestrictionSegment,
  keyValues: DrawingKeys,
  rotate: boolean,
  axe: 'x' | 'y'
) => {
  const rotateTrueValue = axe === 'x' ? 1 : 0;
  const rotateFalseValue = axe === 'x' ? 0 : 1;

  return {
    start: segment[`${keyValues[rotate ? rotateTrueValue : rotateFalseValue]}_start`],
    middle: segment[`${keyValues[rotate ? rotateTrueValue : rotateFalseValue]}_middle`],
    end: segment[`${keyValues[rotate ? rotateTrueValue : rotateFalseValue]}_end`],
  };
};

  const xValues = getValues(electricalConditionSegment, keyValues, rotate, 'x');
  const yValues = getValues(electricalConditionSegment, keyValues, rotate, 'y');

```